google/cloud/firestore_v1/_helpers.py

```diff
@@ -397,8 +401,16 @@ def decode_dict(value_fields, client) -> dict:
         of native Python values converted from the ``value_fields``.
     """
     value_fields_pb = getattr(value_fields, "_pb", value_fields)
+    res = {}
+
+    for key, value in value_fields_pb.items():
+        res[key] = decode_value(value, client)
+
+    if len(res) == 2 and "__type__" in res:
```

Comment:
I'm not sure I like the `len(res) == 2` part. Are we sure that won't ever change? Is it really needed?

It feels safer to cut that part out, and just check for res["__type__"] == "__vector__"

Reply:
SGTM